### ISSUE TITLE
Using string enums with string formatting may cause false positives

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#7563]
This code:
```python
from enum import Enum

class Responses(str, Enum):
    NORMAL = 'something'
    TEMPLATED = 'insert {value} here'

Responses.TEMPLATED.format(value=42)
```
fails with `Not all arguments converted during string formatting`. Using this as `Responses.TEMPLATED.value.format(value=42)` fixes the issue. Both ways actually work with string enums at runtime (only the latter works with normal enums).

**Comment by roberfi** [comment]
The same is happening for 3.11 `StrEnum` classes:
```python
from enum import StrEnum

class Responses(StrEnum):
    NORMAL = "something"
    TEMPLATED = "insert {value} here"

Responses.TEMPLATED.format(value=42)
```
```
tmp.py:7: error: Not all arguments converted during string formatting  [str-format]
Found 1 error in 1 file (checked 1 source file)
```
